### ISSUE TITLE
No vertical scroll in FF for Adaptation: Implementation status - only the first one is visible

### ISSUE THREAD

**Issue by Andy2No** [christofmuc/KnobKraft-orm#208]
I hadn't looked at it for a while, but the *Adaptation: Implementation status* page has no vertical scroll bar in FireFox, for the table, so only the first row of the table is visible, and the rest of it can't be scrolled into view.

The last time I looked at it, it was one big flat page, not a box inside another box, IIRC, so there wasn't a need for that missing scroll bar.

See attached screenshot.

![No vertical scroll in FF for Adaptation Implementation status - only the first one is visible](https://user-images.githubusercontent.com/8021302/212647783-72ef3e8f-8f6e-478d-9321-2f9d17cb4455.gif)

**Comment by christofmuc** [comment]
Confirmed bug, I must have added the file after running the tests for the Behringer RD-8 only. Funny me.

**Comment by christofmuc** [comment]
Workaround, use the history function to access the prior version:

https://github.com/christofmuc/KnobKraft-orm/blob/20aea14236dca045060658023036e6392f49d90c/adaptions/implementation_overview.md

**Comment by Andy2No** [comment]
Thanks.  

I probably ought to learn more about how to use github, but whenever I try, I come across documentation that's just a lot of waffle, instead of concrete instructions.  

I've now had a brief look at the history of that page, and found the user interface pretty hard to follow, e.g. having gone back in time, I see no way to go forwards again, other than starting from scratch.  Maybe that's a fundamental law of github physics? - time only flows in one direction, and that's backwards.

**Comment by christofmuc** [comment]
There is a "history" clickable text at the top of that page, that allows you to access each version of that file. But I agree, UX is not ideal.

**Comment by christofmuc** [comment]
The table is back, though not in colour as you would see it when viewed outside of github.

**Comment by Andy2No** [comment]
Thanks.  Are you able to zoom out enough to eliminate the horizontal scroll bar?  I can't, but maybe it's because I have my system font size (in Win 7) set to "Larger 150%".  It's not really practical for me to set it smaller than that.

It's quite hard to see which box belongs to which synth, and which column, so being able to see the whole width of the table at once would help a little.  I suppose more columns might get added at some point though, bringing the problem back.

**Comment by Andy2No** [comment]
I can work around it by setting up a custom style, in the "Stylus" Firefox plugin, to override the text size used by markdown, but that applies to the whole of github.  For example, I can change to this, from a current font size of 16:

.markdown-body {
	font-family: -apple-system,BlinkMacSystemFont,"Segoe UI","Noto Sans",Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji";
	font-size: 14px;    /* was 16px; */
	line-height: 1.5;
	word-wrap: break-word;
}

Then zooming out to 70% size, in Firefox allows me to eliminate the horizontal scroll bar, and show the full width of the table.   Maybe there's a way to set the font size for the table, separate from Markdown content as a whole?

**Comment by Andy2No** [comment]
Changing the line height to 1 helps too - it allows more rows to be shown at once, without making the table seem overcrowded.

**Comment by Andy2No** [comment]
Is it possible to add a split, like in an Excel spreadsheet?  It would be useful if the body of the table could scroll vertically while leaving the headings row as it is, so we can still see what each box means.

I don't know what the limitations are, but maybe a fixed height table with overflow set to auto (allowing scrolling) for the body of the table would work, with a separate single row table to make the header / table headings). separate from it.

**Comment by christofmuc** [comment]
I think we have very little control over how HTML pages are displayed within the github UI... maybe the best would be to host it somewhere else...

**Comment by christofmuc** [comment]
Python could generate an Excel file instead of a markdown file I guess...

**Comment by Andy2No** [comment]
I didn't see your replies, @christofmuc I had trouble finding this thread again too.  

Yes, it seems they've gone out of their way to make it difficult to present a table well.   Apparently we could add lots of `<sub><sup></sup></sub>` tags, quite probably to each table cell, but we're not really supposed to:

https://stackoverflow.com/questions/66380290/how-to-use-smaller-font-size-in-a-github-table

Adding a link to download an Excel version of the table would be useful.  Ideally, not just CSV, so it can have a split for the column headings, and maybe colour coded, or alternately coloured columns, to help the eye keep track.  

I suppose a .csv export could be added too, but probably everyone can view an Excel file, somehow.  I mostly still use .xls, but there's a Microsoft viewer I can use for .xlsx.

**Comment by markusschloesser** [comment]
I am not in support of a file based table, too much hassle.
How about a link to Google Sheets? Can be used by anyone with just a browser? I'm not sure, but might even support embedding

**Comment by christofmuc** [comment]
@markusschloesser The table is generated by the code, so this is produced as a file that I commit to the repo. Not sure how to do this with a Google doc.

@Andy2No I think I could produce both, the md version we have right now (which, when downloaded and viewed locally, is very useable!), and a csv version for easy import into Google Docs or Excel if somebody wants to work with the document.

The thread is hard to find because the ticket is closed, I think - let me open a new Epic for the documentation!
